connectors/utils.py

```diff
@@ -994,3 +994,101 @@ def get(self, key) -> int:
 
     def to_dict(self):
         return deepcopy(self._storage)
+
+        pass
+
+
+class TooManyErrors(Exception):
+    pass
+
+
+class ErrorMonitor:
+    def __init__(
+        self,
+        max_total_errors=1000,
+        max_consecutive_errors=10,
+        max_error_rate=0.15,
+        error_window_size=100,
+        error_queue_size=20,
```

Comment:
Can we get these configurable through YAML?

Reply:
Yes, absolutely!